### PR TITLE
patch puzzle mutation

### PR DESCRIPTION
**Pull request by lumy**
Issue #25

### CODE REVIEW

File: puzzle.py [hunk 1]
```diff
@@ -103,31 +100,20 @@ def mutate_position(self, ind):
...
   def mutate(self):
     mutation_counter = 0
-    # CONST RAND RATE <!> TO UPDATE WHEN RAND RATE IMPLEMENTED
-    rand_rate = config.mutate_inpd
     for i, ind in enumerate(self.population):
       if i not in config.corner_pos and i not in config.border_pos:
-        rand = random.uniform(0.000, 100.000)
-        if (rand <= rand_rate):
+        rand = random.uniform(0, 100)
+        if (rand <= config.mutate_inpd):
```
Comment (Grenadingue) [inline puzzle.py]: `rand` has range `[0, 100]`, but `config.mutate_inpd` from [`config.py`](https://github.com/lumy/EternityII/blob/%2325_patch_mutation/config.py#L28) has range `[0.0, 1.0]`

File: puzzle.py [hunk 2]
```diff
@@ -103,31 +100,20 @@ def mutate_position(self, ind):
...
           mutation_counter += 1
-          operation = self.choose_mutation(ind)
-          if (random.uniform(0.000, 100.000) <= rand_rate):
-            if (operation == 1):
-              self.mutate_position(ind)
-            else:
-              self.mutate_rotation(ind)
+          operation = random.randint(1, 3)
+          # operation == 3 both if work, operation == 2 only the first one and for operation == 1 only the second one.
```
Comment (Grenadingue) [inline puzzle.py]: It took for me quite a lot of time to understand what means this comment. I needed to understand the function to understand the comment. Comment which is supposed to explain the function. Do you mind if we replace the comment by the following ?
```python
# operation == 1 mutate rotation # operation == 2 mutate position # operation == 3 mutate both
`

**Review by Grenadingue — CHANGES_REQUESTED** [review]

File: puzzle.py (hunk 2 above)
Comment (lumy) [reply puzzle.py]: No probleme. doing that asap

**Comment by lumy** [comment]
Done. need new review

**Review by Grenadingue — APPROVED** [review]

**Review by AdriVig — APPROVED** [review]

**Pull request merged**